### PR TITLE
Credential Registry testcases, help page new test cases added , MOSIP-30054 MOSIP-30230 MOSIP-30282

### PR DESCRIPTION
**Pull request by anup-nehe**
https://mosip.atlassian.net/browse/MOSIP-30282
https://mosip.atlassian.net/browse/MOSIP-30054
https://mosip.atlassian.net/browse/MOSIP-30230
added new test cases:
-Credential Registry testcases.
-valid context in the help page.
-accessing help page in offline.
-scrolling the help page.
-deleteDownloadedVcInOfflineMode.

### CODE REVIEW

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/CredentialRegistryPage.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        clickOnElement(CancelButton);
+        return this;
+    }
+    
+    public CredentialRegistryPage clickOnBackArrow() {
+        clickOnElement(BackArrow);
+        return this;
+    }
+    
+    public String CheckEnvNotChanged() {
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/CredentialRegistryPage.java:52]: camel casing

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/DetailedVcViewPage.java [hunk 2]
```diff
@@ -120,4 +130,18 @@ public HomePage clickOnCrossIcon(){
...
+        clickOnElement(qrCloseIcon);
+        return new HomePage(driver);
+    }
+
+    public PleaseConfirmPopupPage clickOnQrCodeButton(){
+        clickOnElement(detailedVcViewPageQr);
+        return new PleaseConfirmPopupPage(driver);
+    }
+
+    public boolean isQRCodeDisplayed() {
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/DetailedVcViewPage.java:144]: camel casing

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/HassleFreeAuthenticationPage.java [hunk 3]
```diff
@@ -14,6 +14,9 @@ public class HassleFreeAuthenticationPage extends BasePage {
     @AndroidFindBy(xpath = "(//*[@class='android.widget.TextView'])[3]")
     @iOSXCUITFindBy(xpath = "//*[contains(@value,'Authenticate yourself')]")
     private WebElement hassleFreeAuthenticationDescription;
+    
+    @AndroidFindBy(xpath = "//*[contains(@text,'Go Back')]")
+    public WebElement GoBackButton;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/HassleFreeAuthenticationPage.java:19]: camel casing

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/MoreOptionsPage.java [hunk 4]
```diff
@@ -27,10 +27,13 @@ public class MoreOptionsPage extends BasePage {
     @iOSXCUITFindBy(accessibility = "profileAuthenticated")
     private WebElement activatedForOnlineLoginButton;
     
-    @AndroidFindBy(xpath = "//*[@resource-id=\"iconIcon\"]")
+    @AndroidFindBy(uiAutomator = "new UiSelector().resourceId(\"iconIcon\")")
     @iOSXCUITFindBy(iOSClassChain ="**/XCUIElementTypeOther[`label == \"\uE5CD\"`][1]")
     private WebElement CloseButton;
 
+    @AndroidFindBy(uiAutomator = "new UiSelector().textContains(\"Something is wrong. Please try again later!\")")
+    public WebElement SomthingIsWrongPopup;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/MoreOptionsPage.java:35]: camel casing

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/MoreOptionsPage.java [hunk 5]
```diff
@@ -27,10 +27,13 @@ public class MoreOptionsPage extends BasePage {
     @iOSXCUITFindBy(accessibility = "profileAuthenticated")
     private WebElement activatedForOnlineLoginButton;
     
-    @AndroidFindBy(xpath = "//*[@resource-id=\"iconIcon\"]")
+    @AndroidFindBy(uiAutomator = "new UiSelector().resourceId(\"iconIcon\")")
     @iOSXCUITFindBy(iOSClassChain ="**/XCUIElementTypeOther[`label == \"\uE5CD\"`][1]")
     private WebElement CloseButton;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/MoreOptionsPage.java:32]: camel casing

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java [hunk 6]
```diff
@@ -52,6 +52,21 @@ public class SettingsPage extends BasePage {
     @iOSXCUITFindBy(accessibility = "injiTourGuide")
     private WebElement receivedCards;
 
+    @AndroidFindBy(xpath = "//*[contains(@text,'Credential Registry')]")
+    public WebElement clickOnCredentialRegistry;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java:56]: Rename locator to credentialRegistryText

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java [hunk 7]
```diff
@@ -52,6 +52,21 @@ public class SettingsPage extends BasePage {
...
+
+    @AndroidFindBy(xpath = "//*[contains(@text,'Receive Card')]")
+    public WebElement clickOnReceiveCard;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java:59]: Rename locator

File: injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java [hunk 8]
```diff
@@ -52,6 +52,21 @@ public class SettingsPage extends BasePage {
...
+
+    @AndroidFindBy(xpath = "//*[contains(@text,'Tumanggap ng Card')]")
+    public WebElement clickOnReceiveCardInfilipinoLanguage;
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/pages/SettingsPage.java:62]: Rename locator

File: injitest/src/main/java/io/mosip/test/mob/inji/testcases/CredentialRegistry.java [hunk 9]
```diff
@@ -0,0 +1,383 @@
...
+
+        assertTrue(settingsPage.isSettingPageLoaded(), "Verify if setting page is displayed");
+        credentialRegistryPage.clickOnBackArrow();
+        
+        AddNewCardPage addNewCardPage = homePage.downloadCard();
+        assertTrue(homePage.isNoInternetConnectionDisplayed(), "Verify if no internet connection is displayed");
+	}
+	
+	@Test
+	public void GenerateUinOrVidInNewEnv() throws InterruptedException {
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/testcases/CredentialRegistry.java:157]: camel casing

File: injitest/src/main/java/io/mosip/test/mob/inji/testcases/CredentialRegistry.java [hunk 10]
```diff
@@ -0,0 +1,383 @@
...
+        assertTrue(addNewCardPage.isAddNewCardPageLoaded(), "Verify if add new card page is displayed");
+        RetrieveIdPage retrieveIdPage = addNewCardPage.clickOnDownloadViaUin();
+        
+        assertTrue(retrieveIdPage.isRetrieveIdPageLoaded(), "Verify if retrieve id page is displayed");
+        GenerateUinOrVidPage generateUinOrVidPage = retrieveIdPage.clickOnGetItNowText();
+
+        assertTrue(generateUinOrVidPage.isGenerateUinOrVidPageLoaded(), "Verify if generate uin or vid page page is displayed");
+        OtpVerificationPage otpVerification = generateUinOrVidPage.enterApplicationID(TestDataReader.readData("aid")).clickOnGetUinOrVidButton();
+
+        assertTrue(retrieveIdPage.isAidIsNotReadyYetErrorDisplayed(), "Verify if aid is not ready displayed");
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/testcases/CredentialRegistry.java:250]: review and fix assert for UIN verification

File: injitest/src/main/java/io/mosip/test/mob/inji/testcases/NoNetworkAndroidTest.java [hunk 11]
```diff
@@ -153,4 +161,191 @@ public void verifyListOfLanguagesInOfflineMode() {
...
+        
+        ReceiveCardPage receiveCardPage = settingsPage.clickOnReceiveCard();
+        assertTrue(receiveCardPage.isReceiveCardHederDisplayed(), "Verify if QR code  header is displayed");
+        
+        assertTrue(receiveCardPage.isWaitingForConnectionDisplayed(), "Verify if waiting for connection displayed");
+        CommonMethods.disableAirplaneMode();
+    }
+    
+    @Test
+    public void DownloadCardWithoutInternetRetryWithInternet() throws InterruptedException {
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/testcases/NoNetworkAndroidTest.java:318]: camel casing

File: injitest/src/main/java/io/mosip/test/mob/inji/testcases/NoNetworkAndroidTest.java [hunk 12]
```diff
@@ -153,4 +161,191 @@ public void verifyListOfLanguagesInOfflineMode() {
...
+        HomePage homePage = confirmPasscode.confirmPasscode(TestDataReader.readData("passcode"), target);
+
+        assertTrue(homePage.isHomePageLoaded(), "Verify if home page is displayed");
+        AddNewCardPage addNewCardPage = homePage.downloadCard();
+
+        assertTrue(homePage.isNoInternetConnectionDisplayed(), "Verify if no internet connection is displayed");
+        assertTrue(homePage.isTryAgainButtonDisplayed(), "Verify if Try again button displayed");
+
+        CommonMethods.disableAirplaneMode();
+        assertTrue(homePage.isTryAgainButtonDisplayed(), "Verify if Try again button displayed");
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/testcases/NoNetworkAndroidTest.java:345]: Create new method isTryAgainButtonNotDisplayed()

File: injitest/src/main/java/io/mosip/test/mob/inji/testcases/NoNetworkAndroidTest.java [hunk 13]
```diff
@@ -153,4 +161,191 @@ public void verifyListOfLanguagesInOfflineMode() {
...
+        Thread.sleep(3000);
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/testcases/NoNetworkAndroidTest.java:346]: remove static wait

File: injitest/src/main/java/io/mosip/test/mob/inji/testcases/NoNetworkAndroidTest.java [hunk 14]
```diff
@@ -153,4 +161,191 @@ public void verifyListOfLanguagesInOfflineMode() {
...
+        
+        homePage.clickOnTryAgainbutton();
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/testcases/NoNetworkAndroidTest.java:348]: Camel casing

File: injitest/src/main/java/io/mosip/test/mob/inji/testcases/ReceiveCardTest.java [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+import io.mosip.test.mob.inji.pages.ReceiveCardPage;
+import io.mosip.test.mob.inji.pages.SetPasscode;
+import io.mosip.test.mob.inji.pages.SettingsPage;
+import io.mosip.test.mob.inji.pages.WelcomePage;
+import io.mosip.test.mob.inji.utils.TestDataReader;
+
+public class ReceiveCardTest extends BaseTest {
+	
+	 @Test
+	    public void verifyRecivedCard() {
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/testcases/ReceiveCardTest.java:20]: changeMethodName

File: injitest/src/main/java/io/mosip/test/mob/inji/testcases/ReceiveCardTest.java [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+	        assertTrue(homePage.isHomePageLoaded(), "Verify if home page is displayed");
+	        SettingsPage settingsPage = homePage.clickOnSettingIcon();
+	        
+	        ReceiveCardPage receiveCardPage = settingsPage.clickOnReceiveCard();
+	        assertTrue(receiveCardPage.isReceiveCardHederDisplayed(), "Verify if QR code  header is displayed");
+	        assertTrue(receiveCardPage.isWaitingForConnectionDisplayed(), "Verify if waiting for connection displayed");
+	 }
+	 
+	 @Test
+	    public void verifyRecivedCardInFilipinoLanguage() {
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/testcases/ReceiveCardTest.java:47]: rename method name

File: injitest/src/main/java/io/mosip/test/mob/inji/testcases/VcDownloadAndVerifyUsingVidTest.java [hunk 17]
```diff
@@ -134,4 +134,34 @@ public void verifyInvalidVidErrorMessage() {
         assertTrue(retrieveIdPage.isIncorrectInputFormatErrorMessageDisplayed());
     }
     
-}
+    @Test
+    public void downloadAndVerifyVcUsingRevokevid() {
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/testcases/VcDownloadAndVerifyUsingVidTest.java:138]: camelCasing

File: injitest/src/main/java/io/mosip/test/mob/inji/utils/CommonMethods.java [hunk 18]
```diff
@@ -25,28 +25,40 @@ public static void invokeAppFromBackGroundAndroid() {
...
+            String osName = System.getProperty("os.name");
+            if (osName.contains("Windows")) {
+                ProcessBuilder processBuilderAndroid = new ProcessBuilder("cmd.exe", "/c", "adb shell cmd connectivity airplane-mode disable");
+                processBuilderAndroid.redirectErrorStream(true);
+                processBuilderAndroid.start();
+            } else {
+                ProcessBuilder processBuilderOtherOS = new ProcessBuilder("/bin/bash\", \"-c\", \"adb shell cmd connectivity airplane-mode disable");
+                processBuilderOtherOS.redirectErrorStream(true);
+                processBuilderOtherOS.start();
+            }
```
Comment (kamalsinghthoughtworks) [inline injitest/src/main/java/io/mosip/test/mob/inji/utils/CommonMethods.java:59]: remove duplicate code

**Review by lsivanand — APPROVED** [review]

**Review by kamalsinghthoughtworks — APPROVED** [review]

**Review by swatigoel — APPROVED** [review]

**Review by swatigoel — APPROVED** [review]

**Pull request merged**